src/multivariate/mvdiscretenonparametric.jl

```diff
@@ -0,0 +1,153 @@
+probabilities `p`. If the probability vector argument is not passed, then
+equal probability is assigned to each entry in the support.
+
+# Examples
+```julia
+# the rows correspond to the samples
+using LinearAlgebra
+μ = MvDiscreteNonParametric(rand(10,3), normalize!(rand(10),1))
+```
+"""
+function MvDiscreteNonParametric(
+    support::Matrix{<:Real},
+    p::AbstractVector{<:Real}=fill(inv(size(support)[1]), size(support)[1])
+)
+    return MvDiscreteNonParametric(nestedview(support'), p)
+end
+
+Base.eltype(::Type{<:MvDiscreteNonParametric{T}}) where T = T
+
+"""
+    support(d::MvDiscreteNonParametric)
+Get a sorted AbstractVector defining the support of `d`.
+"""
+support(d::MvDiscreteNonParametric) = d.support
+
+"""
+    probs(d::MvDiscreteNonParametric)
+Get the vector of probabilities associated with the support of `d`.
+"""
+probs(d::MvDiscreteNonParametric) = d.p
+
+
+# It would be more intuitive if length was the
+# 
+"""
+    length(d::MvDiscreteNonParametric)
+Retunrs the dimension of the mass points (samples).
+It corresponds to `innersize(d.support)[1]`,
+where `innersize` is a function from from ArraysOfArrays.jl.
+"""
+Base.length(d::MvDiscreteNonParametric) = innersize(d.support)[1]
+
+"""
+    size(d::MvDiscreteNonParametric)
+Returns the size of the support as a tuple where
+the first value is the number of points (samples)
+and the second is the dimension of the samples (e.g ℝⁿ).
+It corresponds to `size(flatview(d.support)')`
+where `flatview` is a function from from ArraysOfArrays.jl
+that turns an Array of Arrays into a matrix.
+"""
+Base.size(d::MvDiscreteNonParametric) = size(flatview(d.support)')
+
+function _rand!(rng::AbstractRNG, d::MvDiscreteNonParametric, x::AbstractVector{T}) where T <: Real
+
+    length(x) == length(d) || throw(DimensionMismatch("Invalid argument dimension."))
+    s = d.support
+    p = d.p
+
+    n = length(p)
+    draw = Base.rand(rng, float(eltype(p)))
+    cp = p[1]
+    i = 1
+    while cp <= draw && i < n
+        @inbounds cp += p[i += 1]
+    end
+    for (j, v) in enumerate(s[i])
+        x[j] = v
+    end
+    return x
+end
+
+function _logpdf(d::MvDiscreteNonParametric, x::AbstractVector{T}) where T <: Real
+    s = support(d)
+    p = probs(d)
+    for i in 1:length(p)
+        if s[i] == x
+            return log(p[i])
+        end
+    end
+    return log(zero(eltype(p)))
+end
+
+
+function mean(d::MvDiscreteNonParametric)
+    return StatsBase.mean(d.support, weights(d.p))
+end
+
+function var(d::MvDiscreteNonParametric)
+    x = support(d)
+    p = probs(d)
+    return StatsBase.var(x, Weights(p, one(eltype(p))), corrected=false)
+end
+
+function cov(d::MvDiscreteNonParametric)
+    x = support(d)
+    p = probs(d)
+    return cov(x, Weights(p, one(eltype(p))), corrected=false)
+end
+
+entropy(d::MvDiscreteNonParametric) = entropy(probs(d))
+entropy(d::MvDiscreteNonParametric, b::Real) = entropy(probs(d), b)
```

Comment:
This is defined in StatsBase (https://github.com/JuliaStats/StatsBase.jl/blob/0a179531f14d0140356fcf1f1d55bd0240030cf2/src/scalarstats.jl#L537):
```suggestion
```